wireapp/wire-android-reloaded
# refactor: extract email login into separate package [AR-1159]

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

There is only a single type of login in the app which is the one using email/user handle. We need to add the second method - SSO login so it's necessary to first extract and separate email login with its viewmodel and composables into its own package.

### Solutions

All logic and views related to email login extracted to another package `com.wire.android.ui.authentication.login.email`. The main login screen now will be using `LoginEmail` screen from this package, and in the next PR it will have a tab row with both email and sso login.

### Testing

Added LoginEmailViewModelTest and modified LoginViewModelTest.

#### Test Coverage (Optional)


#### How to Test

Apart from automated tests, you can open login screen and check if anything changed from the UI and UX perspective (it shouldn't 😄 ).

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
